src/ResourceManager/Insights/Commands.Insights/Diagnostics/SetAzureRmDiagnosticSettingCommand.cs

```diff
@@ -31,77 +31,115 @@ namespace Microsoft.Azure.Commands.Insights.Diagnostics
     [Cmdlet(VerbsCommon.Set, "AzureRmDiagnosticSetting", SupportsShouldProcess = true), OutputType(typeof(PSServiceDiagnosticSettings))]
     public class SetAzureRmDiagnosticSettingCommand : ManagementCmdletBase
     {
+        internal const string SetAzureRmDiagnosticSettingOldParamGroup = "OldSetDiagnosticSetting";
+        internal const string SetAzureRmDiagnosticSettingNewParamGroup = "NewSetDiagnosticSetting";
+
         public const string StorageAccountIdParamName = "StorageAccountId";
         public const string ServiceBusRuleIdParamName = "ServiceBusRuleId";
+        public const string EventHubNameParamName = "EventHubName";
         public const string EventHubRuleIdParamName = "EventHubAuthorizationRuleId";
         public const string WorkspacetIdParamName = "WorkspaceId";
         public const string EnabledParamName = "Enabled";
 
+        /// <summary>
+        /// This is a temporary constant to provide backwards compatibility
+        /// </summary>
+        internal const string TempServiceName = "service";
+
         #region Parameters declarations
 
+        /// <summary>
+        /// Gets or sets the InputObject parameter of the cmdlet
+        /// </summary>
+        [Parameter(ParameterSetName = SetAzureRmDiagnosticSettingNewParamGroup, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The complete spec of a DiagnosticSettingSetting")]
+        [ValidateNotNullOrEmpty]
+        public PSServiceDiagnosticSettings InputObject { get; set; }
+
         /// <summary>
         /// Gets or sets the resourceId parameter of the cmdlet
         /// </summary>
         [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The resource id")]
```

Comment:
@pkamath13 @gucalder we're going to need to add `ParameterSetName` properties to these parameters since we shouldn't be able to provide both `-ResourceId` and `-InputObject` in the same cmdlet call

Reply:
I added Name and ResourceId to one of the existing sets and made the adjustments needed for the cmdlet to know the name and the resourceId.